sql/core/src/main/scala/org/apache/spark/sql/catalyst/util/V2ExpressionBuilder.scala

```diff
@@ -234,7 +234,13 @@ class V2ExpressionBuilder(e: Expression, isPredicate: Boolean = false) {
       if (l.isDefined && r.isDefined) {
         b match {
           case _: Predicate =>
-            Some(new V2Predicate(b.sqlOperator, Array[V2Expression](l.get, r.get)))
+            if (l.get.isInstanceOf[LiteralValue[_]] && r.get.isInstanceOf[FieldReference]) {
```

Comment:
+1 for @sunchao 's [comment](https://github.com/apache/spark/pull/37197#discussion_r926290587) for safety. Could you try to check, @huaxingao ?

Reply:
Sorry I missed that comment. Fixed.